src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -456,6 +457,14 @@ private function resolveVariables($value)
                 $value = (string) getenv($name);
             }
 
+            if ('' === $value && isset($matches['default_value'])) {
+                if (false !== strpbrk($matches['default_value'], '\'"{$')) {
+                    throw $this->createFormatException(sprintf('Unsupported character found in the default value of variable "$%s".', $name));
```

Comment:
If not, I think we should embed the matched string here like this:

```php
if (false !== $tokenMismatch = strpbrk($matches['default_value'], '\'"{$')) {
    throw $this->createFormatException(sprintf('Unsupported token "%s" found in the default value of variable "$%s".', $tokenMismatch, $name));
}
```

Reply:
Thanks, even better 👍